webpack.config.mjs

```diff
@@ -105,78 +119,104 @@ const bundleStandalone = {
       }
     ]
   },
+
   entry: {
-    'accessible-autocomplete.min': './wrapper.js'
-  },
-  output: {
-    path: join(cwd(), 'dist'),
-    publicPath: '/dist',
-    filename: '[name].js',
-    library: 'accessibleAutocomplete',
-    libraryExport: 'default',
-    libraryTarget: 'umd'
+    'accessible-autocomplete': {
+      import: join(cwd(), 'src/wrapper.js'),
+      filename: '[name].min.js',
+      library: {
+        export: 'default',
+        name: 'accessibleAutocomplete',
+        type: 'umd'
+      }
+    }
   },
+
   plugins: plugins
     .concat([new webpack.DefinePlugin({
       'process.env.COMPONENT_LIBRARY': '"PREACT"'
     })])
 }
 
+/**
+ * Bundle for Preact 'accessible-autocomplete/preact.js'
+ *
+ * @satisfies {WebpackConfiguration}
+ */
 const bundlePreact = {
   ...config,
+
   entry: {
-    'lib/accessible-autocomplete.preact.min': './autocomplete.js'
-  },
-  output: {
-    path: join(cwd(), 'dist'),
-    publicPath: '/',
-    filename: '[name].js',
-    library: 'Autocomplete',
-    libraryTarget: 'umd'
+    'accessible-autocomplete.preact': {
+      import: join(cwd(), 'src/autocomplete.js'),
+      filename: 'lib/[name].min.js',
+      library: {
+        name: 'Autocomplete',
+        type: 'umd',
+        umdNamedDefine: true
+      }
+    }
   },
+
   externals: {
-    preact: {
-      amd: 'preact',
-      commonjs: 'preact',
-      commonjs2: 'preact',
-      root: 'preact'
-    }
+    preact: 'preact'
   },
+
+  output: {
+    ...config.output,
+    globalObject: 'window'
```

Comment:
**issue** I think this one needs to go in the shared config's `output`, so that the wrapped version of the library also gets added to `window`, as it is with Webpack 4. With Webpack 5, looks like it gets added to `self` instead.

Reply:
Interestingly, looks like on GOV.UK Frontend, we add to `this` rather than either of `window` or `self`. `this` sounds more portable, but:
- isn't like for like with the current output of Webpack
- I'm not quite sure if there'd be unforeseen side effects

Reply:
@romaricpascal Shall we pair on this?

You'll notice the difference when importing into the Review app

Reply:
Yeah, it's when running without a bundler (e.g. dependencies via unpkg `<script>`)

It means we can use `window.React.createElement()` in user code but still ensure React inside our bundled library code still resolves and extends the `window.React` version using webpack `globalObject`

I'll add a comment to explain:

```js
// Support extending `window.React` when not bundled
// e.g. with all dependencies included via unpkg.com
globalObject: 'window'
```